### PR TITLE
KAADO-23: Create user auth service

### PR DESCRIPTION
**Pull request by nasudon**
- Create user auth service
- Create a simple login page
- Create a simple sign up page
- Update user schema

### CODE REVIEW

File: src/app/auth/login/form.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    const searchParams = useSearchParams();
+    const callbackUrl = searchParams.get('callbackUrl') || '/member';
+
+    const onSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        try {
+            setLoading(true);
+            setFormValues({ username: '', password: '' });
+
+            const res = await signIn('credentials', {
```
Comment (tsoibet) [inline src/app/auth/login/form.tsx:25]: I think signIn function should take a credential name, which is defined as `Log in` in `auth/[...nextauth]/route.ts`.

File: src/app/api/auth/[...nextauth]/route.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                    return null;
+                }
+            },
+        }),
+    ],
+    callbacks: {
+        async jwt({ token, account }) {
+            // Persist the OAuth access_token to the token right after signin
+            if (account) {
+                token.userId = account.providerAccountId;
```
Comment (tsoibet) [inline src/app/api/auth/[...nextauth]/route.ts]: I think `providerAccountId` is for OAuth. `account.id` may be better?

File: src/app/auth/login/form.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    const router = useRouter();
+    const [loading, setLoading] = useState(false);
+    const [formValues, setFormValues] = useState({
+        username: '',
+        password: '',
+    });
+    const [errorMsg, setErrorMsg] = useState('');
+
+    const searchParams = useSearchParams();
+    const callbackUrl = searchParams.get('callbackUrl') || '/member';
```
Comment (tsoibet) [inline src/app/auth/login/form.tsx:17]: I think `callbackUrl` is needed only when O-Auth is being used.

https://next-auth.js.org/getting-started/example#configuring-callback-url-oauth-only

File: src/app/auth/login/form.tsx (hunk 1 above)
Comment (nasudon) [reply src/app/auth/login/form.tsx:25]: Checked together and found that the name is the name display on the sign in form only.
ref.: https://next-auth.js.org/providers/credentials#example---username--password

File: src/app/api/auth/[...nextauth]/route.ts (hunk 2 above)
Comment (nasudon) [reply src/app/api/auth/[...nextauth]/route.ts]: Fixed: https://github.com/tsoibet/kaado/pull/11/commits/d0f09d2e4f427b8b73be5fb026bcf4b590665cea

File: src/app/auth/login/form.tsx (hunk 3 above)
Comment (nasudon) [reply src/app/auth/login/form.tsx:17]: The callbackUrl here is for user to redirect to the page that they just logged out from.

**Review by tsoibet — APPROVED** [review]
LGTM

**Pull request merged**